File: Day3_AdvancedTopics/5.Parallel/omp_Sections_Example.f90
PROGRAM VEC_ADD_SECTIONS

  INTEGER N, I, TID, OMP_GET_THREAD_NUM
  PARAMETER (N=1000)
  REAL A(N), B(N), C(N), D(N)

  !Some initializations
  DO I = 1, N
     A(I) = I * 1.5
     B(I) = I + 22.35
  ENDDO

  !$OMP PARALLEL SHARED(A,B,C,D), PRIVATE(I, TID)
  TID=OMP_GET_THREAD_NUM()

  !$OMP SECTIONS

  !$OMP SECTION
  WRITE(*,*) 'Thread: ', TID, ' Working on + section'
  DO I = 1, N
     C(I) = A(I) + B(I)
  ENDDO

  !$OMP SECTION
  WRITE(*,*) 'Thread: ', TID, ' Working on * section'
  DO I = 1, N
     D(I) = A(I) * B(I)
  ENDDO

  !$OMP END SECTIONS NOWAIT

  !$OMP END PARALLEL

END PROGRAM VEC_ADD_SECTIONS

File: Day3_AdvancedTopics/5.Parallel/omp_Sections_Example.c
#include <stdio.h>
#include <omp.h>
#define N 1000

int main(int argc, char *argv[]) {

  int i, tid;
  float a[N], b[N], c[N], d[N];
  
  /* Some initializations */
  for (i=0; i < N; i++) {
    a[i] = i * 1.5;
    b[i] = i + 22.35;
  }

#pragma omp parallel shared(a,b,c,d) private(i, tid)
  {
    tid=omp_get_thread_num();

#pragma omp sections nowait
    {
      
#pragma omp section
      {
      printf("Thread: %d working on + section\n", tid);
      for (i=0; i < N; i++)
	c[i] = a[i] + b[i];
      }

#pragma omp section
      {
      printf("Thread: %d working on * section\n", tid);
      for (i=0; i < N; i++)
	d[i] = a[i] * b[i];
      }

    }  /* end of sections */

  }  /* end of parallel region */

}
